### PR TITLE
Fixed GCC10 + F2008 compliance

### PR DESCRIPTION
**Pull request by nwu63**
## Purpose
I added the `-std=f2003` flag and fixed all the errors that came up. Since I'm on GCC 10, I also fixed the argument mismatch issue.

Note that some warnings still remain:
```
parameterizations.f90:38:61:

                       (X(3, K, J, I) - X(3, K, J, I-1)) ** 2)
                                                             1
Warning: Obsolescent feature: Statement function at (1)
```

As part of the process, I removed the `norm` function definition and replaced it by the F2008 intrinsic `NORM2`. I also replaced one scalar call to `norm` with the `ABS` function which is equivalent for scalars.

## Type of change
What types of change is it?
_Select the appropriate type(s) that describe this PR_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [x] Maintenance update
- [ ] Other (please describe)

## Testing
I updated the config file so the `-std=f95` code will be tested, but since we are on GCC 9 I guess we should just observe that the argument mismatch warnings should no longer show up in the compile log.

## Checklist
_Put an `x` in the boxes that apply._

- [ ] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [x] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

File: src/projections.F90
Comment (nwu63) [inline src/projections.F90:59]: Is this an okay solution? I create a temporary singleton array and use this to pass value to the subroutine, as recommended [here](https://gcc.gnu.org/gcc-10/porting_to.html).

File: src/projections.F90
Comment (nwu63) [inline src/projections.F90:1501]: I'm assuming the missing parameter here is 3, please double check.

File: src/projections.F90
Comment (bernardopacini) [reply src/projections.F90:59]: This seems okay to me.

File: src/projections.F90
Comment (bernardopacini) [reply src/projections.F90:1501]: I am having trouble figuring out where this `norm()` function is. Do you know where it references?

File: config/defaults/config.LINUX_GFORTRAN.mk
Comment (bernardopacini) [inline config/defaults/config.LINUX_GFORTRAN.mk:8]: This is likely not within the scope of this PR, but I have read that it is bad practice to use the `-fdefault-real-8` flag. It may not matter for us, but I want to note it.

**Review by bernardopacini — COMMENTED** [review]
Besides the three comments, the rest looks good to me.

File: src/projections.F90
Comment (eirikurj) [reply src/projections.F90:59]: In general I am not a fan, but this is the simplest. Would need to think for a better solution.

File: src/adtProjections.F90
Comment (eirikurj) [inline src/adtProjections.F90:1290]: I almost feel that this is a step backwards. Since pyspline does not require tapenade I would be more in favor of using 2003/2008.

File: src/projections.F90
Comment (eirikurj) [reply src/projections.F90:1501]: The norm function is defined in `compute_curve.f90`. With 2008 we could replace this with `norm2`. Maybe in a separate PR.

**Review by eirikurj — CHANGES_REQUESTED** [review]

File: src/projections.F90
Comment (nwu63) [reply src/projections.F90:1501]: Right, I would prefer a separate PR if we want to update to modern intrinsics.

File: src/adtProjections.F90
Comment (nwu63) [reply src/adtProjections.F90:1290]: Let me update to F2008 then, and maybe update the intrinsics at the same time.

File: src/projections.F90
Comment (nwu63) [reply src/projections.F90]: Never mind I just did it here

File: src/projections.F90
Comment (nwu63) [reply src/projections.F90]: I removed this and replaced by the `norm` invocation with `ABS` which is equivalent for scalars.

File: config/defaults/config.LINUX_GFORTRAN.mk
Comment (nwu63) [reply config/defaults/config.LINUX_GFORTRAN.mk]: @eirikurj do you know anything about this? Why is it bad practice? We use it pretty much everywhere I think

**Comment by nwu63** [comment]
Okay I updated the following
- switched to F2003
- reverted the subroutine name shortening
- switched to using NORM2 intrinsic
- replaced scalar invocation of norm with `ABS`

There are other things that can be replaced by intrinsics, such as the `dotproduct` subroutine, but maybe we should save that for later. A lot of our codes can be modernized though.

**Comment by bernardopacini** [comment]
These new updates look good to me, but I want to wait for @eirikurj's approval before merging.

**Review by joanibal — APPROVED** [review]

File: config/defaults/config.LINUX_GFORTRAN.mk
Comment (eirikurj) [reply config/defaults/config.LINUX_GFORTRAN.mk]: Maybe @bernardopacini can elaborate on this, but I guess this can be considered bad practice by some, since it may not always give the same precision. However, while not perfect, in most of our architectures it should not be an issue and I would argue that this currently safeguards us from potential precision issues. For the current implementation, I would not want to remove it until we have addressed that.

**Review by eirikurj — APPROVED** [review]

File: config/defaults/config.LINUX_GFORTRAN.mk
Comment (bernardopacini) [reply config/defaults/config.LINUX_GFORTRAN.mk]: I do not have anything formal to refer to about why it is bad practice, I had simply read that in a few places. I agree that for us it is likely far more problematic to take this out than to leave it in.

**Review by bernardopacini — APPROVED** [review]

**Pull request merged**
